Hw5.py

```diff
@@ -0,0 +1,74 @@
+import itertools
+import sys
+
+import mariadb
+import pandas as pd
+import plotly.express as px
+
+
+def __main__():
+    try:
+        conn = mariadb.connect(
+            user="root",
+            password="root",  # pragma: allowlist secret
+            host="localhost",
+            port=3306,
+            database="baseball",
+        )
+    except mariadb.Error as e:
+        print(f"Error connecting to MariaDB Platform: {e}")
+        sys.exit(1)
+
+    # Get Cursor
+    cur = conn.cursor()
+    cur.execute(
+        """
+        Select * from batter_total
+    """
+    )
+
+    df = pd.DataFrame(cur.fetchall())
+    print(df)
+
+    response = ["win_lose"]
+    cat_col = []
+    cont_col = []
+    cat_response = []
+    cont_response = []
+
+    # important loop that determines categorical variables.
+    for i in df:
+        if len(df[i].unique()) < 10:
+            cat_col.append(i)
+        elif len(df[i].unique()) > 10:
+            cont_col.append(i)
+
+    for z in response:
+        if z in cat_col:
+            cat_response.append(z)
+            cat_col.remove(z)
+        elif z in cont_col:
+            cont_response.append(z)
+            cont_col.remove(z)
+
+    cont_cont_pair = list(itertools.combinations(cont_col, 2))
+
+    oo_df = pd.DataFrame(index=cont_col, columns=cont_col)
+
+    for c, d in cont_cont_pair:
```

Comment:
not sure why you are doing it this way. Is it so you only get unique pairs (hr/bat_avg and not bat_avg/hr)? If so I would look into masking. I was able to run your code by removing this loop and doing:
corr_cont = df.corr()
oo_df2 = corr_cont.stack....